js/src/scrollspy.js

```diff
@@ -133,7 +133,7 @@ const ScrollSpy = (($) => {
             target = $(targetSelector)[0]
           }
 
-          if (target && (target.offsetWidth || target.offsetHeight)) {
+          if (target && (target.getBoundingClientRect().width || target.getBoundingClientRect().height)) {
```

Comment:
I found something about the algorithm if it can help you : https://drafts.csswg.org/cssom-view/#dom-element-getboundingclientrect